File: apiEZPZ/src/main/java/com/example/apiezpz/search/repository/SearchHistoryRepository.java
package com.example.apiezpz.search.repository;

import java.time.LocalDateTime;
import java.util.List;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import com.example.apiezpz.auth.entity.User;
import com.example.apiezpz.search.entity.SearchHistory;

import jakarta.transaction.Transactional;

@Repository
public interface SearchHistoryRepository extends JpaRepository<SearchHistory, Long> {
    List<SearchHistory> findByUserOrderBySearchDateDesc(User user);
    List<SearchHistory> findByCategoryOrderBySearchDateDesc(String category);
    List<SearchHistory> findByUserAndSearchDateBetweenOrderBySearchDateDesc(User user, LocalDateTime startDate, LocalDateTime endDate);
    
    List<SearchHistory> findByUserAndSearchDateAfterOrderBySearchDateDesc(User user, LocalDateTime startDate);
    
    List<SearchHistory> findByUserAndKeywordContainingOrderBySearchDateDesc(User user, String keyword);
    
    List<SearchHistory> findByUserAndKeywordIsNotNullAndKeywordNotOrderBySearchDateDesc(User user, String emptyString);
    
    List<SearchHistory> findByUserAndSearchDateBetweenAndKeywordIsNotNullAndKeywordNotOrderBySearchDateDesc(
            User user, LocalDateTime startDate, LocalDateTime endDate, String emptyString);

    @Transactional
    void deleteByUser(User user);
}

File: backend/src/main/kotlin/com/example/apiezpz/search/repository/SearchHistoryRepository.kt
package com.example.apiezpz.search.repository

import com.example.apiezpz.auth.entity.User
import com.example.apiezpz.search.entity.SearchHistory
import jakarta.transaction.Transactional
import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.time.LocalDateTime

@Repository
interface SearchHistoryRepository : JpaRepository<SearchHistory, Long> {
    fun findByUserOrderBySearchDateDesc(user: User): List<SearchHistory>
    fun findByCategoryOrderBySearchDateDesc(category: String): List<SearchHistory>
    fun findByUserAndSearchDateBetweenOrderBySearchDateDesc(user: User, startDate: LocalDateTime, endDate: LocalDateTime): List<SearchHistory>
    
    fun findByUserAndSearchDateAfterOrderBySearchDateDesc(user: User, startDate: LocalDateTime): List<SearchHistory>
    
    fun findByUserAndSearchDateAfter(user: User, startDate: LocalDateTime, pageable: Pageable): Page<SearchHistory>
    
    fun findByUserAndKeywordContainingOrderBySearchDateDesc(user: User, keyword: String): List<SearchHistory>
    
    fun findByUserAndKeywordIsNotNullAndKeywordNotOrderBySearchDateDesc(user: User, emptyString: String): List<SearchHistory>
    
    fun findByUserAndSearchDateBetweenAndKeywordIsNotNullAndKeywordNotOrderBySearchDateDesc(
        user: User, startDate: LocalDateTime, endDate: LocalDateTime, emptyString: String
    ): List<SearchHistory>

    @Transactional
    fun deleteByUser(user: User)
} 